modules/ui/fields/localized.js

```diff
@@ -122,7 +122,10 @@ export function uiFieldLocalized(field, context) {
         var existingLangs = new Set(existingLangsOrdered.filter(Boolean));
 
         for (var k in tags) {
-            var m = k.match(/^(.*):(.*)$/);
+            // matches for field:<code>, where <code> is a BCP 47 locale code
+            // motivation is to avoid matching on similarly formatted tags that are
+            // not for languages, e.g. name:left, name:source, etc.
+            var m = k.match(/^(.*):([A-Za-z]{2,3}(?:-[A-Za-z]{4})?)$/);
```

Comment:
Oh, right, I missed the hyphens in my earlier suggestion. This would be more correct:

```suggestion
            var m = k.match(/^(.*):([a-z]{2,3}(?:-[A-Z][a-z]{3})?(?:-[A-Z]{2})?)$/);
```

This will take care of country codes, and the capitalization matches the preferred case in the standards [as documented here](https://wiki.openstreetmap.org/wiki/Special:PermanentLink/2322836#Issues). Other cases are quite rare these days. You can get a sense for the prevalence of certain name subkeys [using taginfo](https://taginfo.openstreetmap.org/search?q=name%3A).

Nonstandard or deprecated keys will continue to show up in the raw tags section of the sidebar, but they won’t show up in the Multilingual Names field. It probably wouldn’t have been a great experience anyways, with the language textbox showing the raw nonstandard code instead of the name of the language.

StreetComplete uses the localized name keys somewhat differently than iD does, so I think a slight discrepancy in the regular expressions should be OK.